internet-indicator@sangorys/files/internet-indicator@sangorys/applet.js

```diff
@@ -0,0 +1,291 @@
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // CALLBACK WHEN SETTINGS ARE CHANGED => WARNING THIS CALLBACK IS NEVER CALLED WHEN THE CONFIG IS CHANGED
+    set_update_interval_when_no_internet(){
+    	global.log("update_interval_when_ninternet=" + this.update_interval_when_internet);
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _new_freq(){
+    	global.log(this.update_interval);
+        if (this._updateLoopID) {
+			Mainloop.source_remove(this._updateLoopID);
+		}
+        this._update_loop();
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _get_status(){
+        let status = this._run_cmd("nordvpn status");
+        let result = status.split("\n")[0].split(": ")[1];
+        let outString;
+        if (result === "Connected"){
+            this.connected = true;
+            outString = "ON";
+        }else if (result === "Disconnected"){
+            this.connected = false;
+            outString = "OFF";
+        }else{
+            this.connected = false,
+            outString = "..."
+        }
+        this.set_applet_label(outString);
+    }
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _update_loop() {
+		this._send_ping();	
+		//Util.spawnCommandLine("/usr/sbin/notify-send Applet '" + this.update_interval + "'"); //OK
+		//global.logError(this.update_interval);
+		this._updateLoopID = Mainloop.timeout_add(this.update_interval * 1000, Lang.bind(this, this._update_loop));
+	}
+
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    _send_ping() {
+        
+		// INIT
+		let [result, stdout, stderr] = ["", "", ""];
+		nbTry = nbTry + 1;
+		//Util.spawnCommandLine("/usr/sbin/notify-send Applet 'Scan " + nbTry + "'");
+		printDebug("Scan " + nbTry + " (period=" + this.update_interval + "s)");
+
+		
+        //////////////////////////////////////////////
+        // RUN PING
+		try {
+			[result, stdout, stderr] = GLib.spawn_command_line_sync("ping www.google.fr -c 1 -w 1000");
+			//printDebug("result=" + result);
+			//printDebug("stderr=" + stderr);
+			//printDebug("stdout=" + stdout);
+			//printDebug("oldStderr=" + oldStderr);
+		  }
+		catch (e) {
+			global.logError(e);
+		}
+		
+        
+        //////////////////////////////////////////////
+        // ANALYZE THE PING ANSWER
+		//global.logError("analyse");  
+		//global.logError(stdout.toString());  
+		//global.logError(stderr.toString().length);
+		
+		if (stderr == "")
+		{
+			//global.logError("ping OK");  
+			if (stderr.toString() != oldStderr)
+			{
+			  //global.logError("connecté");
+			  oldStderr = stderr.toString();
+			  Util.spawnCommandLine("/usr/sbin/notify-send 'Internet OK'");
```

Comment:
You are sending an untranslated string to notify

Reply:
Yes, this is something I want to improve but I don't know yet how to use "language dictionnary" to have my applets translated for each language.
Can you help me with an example, a tuto or a simple explanation ?

Reply:
As far as I know there is no pre-injected function named `_()` so you need to write your own. There is a chance I'm wrong tho, so do this last.

You did generate the translations correctly, but you will need to install it as well to test it.
You can do that with `cinnamon-spices-makepot "internet-indicator@sangorys" --install`

Reply:
You are great support ! Thanks.

Translation now works.

I guess everything works now for a 1 version. No ?